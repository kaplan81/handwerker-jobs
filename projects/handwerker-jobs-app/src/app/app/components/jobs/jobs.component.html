<mat-toolbar color="primary">
  <span class="header-title">{{ title }}</span>
</mat-toolbar>

<mat-card class="shipping-card">
  <mat-card-header>
    <mat-card-title>Jobs</mat-card-title>
  </mat-card-header>
  <mat-card-content>
    <mat-accordion>
      <mat-expansion-panel *ngFor="let job of jobs">
        <mat-expansion-panel-header>
          <mat-panel-title> {{ job.title }} ({{ job.zip_code }} {{ job.city }})</mat-panel-title>
        </mat-expansion-panel-header>
        <mat-panel-description>
          {{ job.description }}
        </mat-panel-description>
      </mat-expansion-panel>
    </mat-accordion>
  </mat-card-content>
</mat-card>
