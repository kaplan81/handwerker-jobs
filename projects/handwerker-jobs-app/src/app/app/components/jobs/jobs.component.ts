import { Component, Input } from '@angular/core';
import * as fromRootModels from '@handwerker-jobs-app/models';

@Component({
  selector: 'hdj-jobs',
  styleUrls: ['./jobs.component.scss'],
  templateUrl: './jobs.component.html'
})
export class JobsComponent {
  @Input() title: string;
  @Input() jobs: fromRootModels.Job[];
}
