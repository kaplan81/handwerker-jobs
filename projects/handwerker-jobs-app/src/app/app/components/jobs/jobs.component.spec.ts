/* tslint:disable: component-selector */
import { Component } from '@angular/core';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { JobsComponent } from '@handwerker-jobs-app/components/jobs/jobs.component';

@Component({ selector: 'mat-toolbar', template: '' })
export class MatToolbarStubComponent {}

@Component({ selector: 'mat-card', template: '' })
export class MatCardStubComponent {}

@Component({ selector: 'mat-card-header', template: '' })
export class MatCardHeaderStubComponent {}

@Component({ selector: 'mat-card-title', template: '' })
export class MatCardTitleStubComponent {}

@Component({ selector: 'mat-card-content', template: '' })
export class MatCardContentStubComponent {}

@Component({ selector: 'mat-accordion', template: '' })
export class MatAccordionStubComponent {}

@Component({ selector: 'mat-expansion-panel', template: '' })
export class MatExpansionPanelStubComponent {}

@Component({ selector: 'mat-expansion-panel-header', template: '' })
export class MatExpansionPanelHeaderStubComponent {}

@Component({ selector: 'mat-panel-title', template: '' })
export class MatPanelTitleStubComponent {}

@Component({ selector: 'mat-panel-description', template: '' })
export class MatExpansionPanelDescriptionStubComponent {}

describe('JobsComponent', () => {
  let fixture: ComponentFixture<JobsComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [
        JobsComponent,
        MatToolbarStubComponent,
        MatCardStubComponent,
        MatCardHeaderStubComponent,
        MatCardTitleStubComponent,
        MatCardContentStubComponent,
        MatAccordionStubComponent,
        MatExpansionPanelStubComponent,
        MatExpansionPanelHeaderStubComponent,
        MatPanelTitleStubComponent,
        MatExpansionPanelDescriptionStubComponent
      ]
    })
      .compileComponents()
      .then(() => {
        fixture = TestBed.createComponent(JobsComponent);
      });
  }));

  it('should match snapshot', () => {
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
