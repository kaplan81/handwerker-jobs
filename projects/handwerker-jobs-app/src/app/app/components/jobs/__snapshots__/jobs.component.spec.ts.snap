// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobsComponent should match snapshot 1`] = `
<hdj-jobs>
  <mat-toolbar
    color="primary"
  /><mat-card
    class="shipping-card"
  />
</hdj-jobs>
`;
